doc/source/serve/deployment-graph-cookbook.md

```diff
@@ -0,0 +1,60 @@
+---
+jupytext:
+  formats: ipynb,md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.13.6
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+(serve-deployment-graph-cookbook)=
+
+# Deployment Graph Cookbook
+
+### Chain Nodes
+
+The example shows how to chain nodes using the same class and having different args passed in
+
++++
+
+```python
+import ray
+from ray import serve
+from ray.experimental.dag.input_node import InputNode
+from ray.serve.drivers import DAGDriver
+
+ray.init(num_cpus=16)
+serve.start()
+
+@serve.deployment
+class Model:
+   def __init__(self, weight):
+      self.weight = weight
+   def forward(self, input):
+      return input +  self.weight
+
+
+# 10 nodes chain
+num_nodes = 10
+nodes = [Model.bind(w) for w in range(num_nodes)]
+prev_outputs = [None] * num_nodes
+with InputNode() as dag_input:
+   for i in range(num_nodes):
+      if i == 0:
+         # first node
+         prev_outputs[i] = nodes[i].forward.bind(dag_input)
+      else:
+         prev_outputs[i] = nodes[i].forward.bind(prev_outputs[i - 1])
+   
+   serve_dag = DAGDriver.options(route_prefix="/my-dag").bind(prev_outputs[-1])
```

Comment:
i would suggest cutting this part completely and directly do `dag.execute(0)` and prints its output in the doc. The sequence of action we want user to take is Ray DAG -> Serve DAG where DAGDriver and serve.run() comes later when user needs http / serving / rolling updates.

Therefore i will be helpful to have sections like "How do I serve my DAG with HTTP" , "How do i add adapter / schema to my http input" and "How do I scale one node in my deployment graph"

Reply:
prev_outputs[-1] is a bit confusing here as cookbook that aims for simplicity :)

Reply:
Do you have suggestions what we should describe this scenario?  I can add comments mention that "we need to use the last node to execute the dag" But let me know if you have good idea